### PR TITLE
fix: Overflows/underflows on deserialization

### PR DESCRIPTION
**Pull request by untitaker**

### CODE REVIEW

File: general/src/macros.rs [hunk 1]
```diff
@@ -39,23 +39,23 @@ macro_rules! numeric_meta_structure {
...
-                    Annotated(Some(Value::F64(value)), meta) => {
-                        Annotated(Some(value as $type), meta)
-                    }
-                    Annotated(Some(Value::Null), meta) => Annotated(None, meta),
-                    Annotated(None, meta) => Annotated(None, meta),
-                    Annotated(Some(value), mut meta) => {
-                        meta.add_unexpected_value_error($expectation, value);
-                        Annotated(None, meta)
+                value.and_then(|value| {
+                    let number: Option<$type> = match value {
```
Comment (jan-auer) [inline general/src/macros.rs]: Just wondering, is this type declaration required?

File: general/src/macros.rs [hunk 2]
```diff
@@ -39,23 +39,23 @@ macro_rules! numeric_meta_structure {
...
+                        Value::U64(x) => num_traits::cast::cast(x),
+                        Value::I64(x) => num_traits::cast::cast(x),
+                        Value::F64(x) => num_traits::cast::cast(x),
+                        _ => None,
+                    };
+
+                    match number {
+                        Some(x) => Ok(x),
```
Comment (jan-auer) [inline general/src/macros.rs]: You could avoid the confusing Result here by returning Annotated directly in both cases

**Review by jan-auer — APPROVED** [review]
LGTM, two nit picks below

File: general/src/macros.rs (hunk 1 above)
Comment (untitaker) [reply general/src/macros.rs]: nah, it was a leftover from when I confused the compiler into confusing me

File: general/src/macros.rs (hunk 2 above)
Comment (untitaker) [reply general/src/macros.rs]: Fixed. I think we need to have a discussion about `IntoAnnotated`, I think some of the things you can convert are counterintuitive in their behavior.

I would want to get rid of it and try to adopt a builder pattern on `Annotated`

**Pull request merged**